hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Alert.java

```diff
@@ -63,6 +63,9 @@
     private List<Note> notes;
 
     @JsonInclude(Include.NON_EMPTY)
+    private List<LifeCycle> lifecycle;
```

Comment:
Storing the lifecycle is needed for two reasons:
- In the model we might have open -> ack -> resolved -> ack -> resolved (transitions from resolved to ack states are valid).
- At the moment that we index resolved/ack dates for filtering with an auxiliar table we need to track the state to maintain this.
So the proper way was to store the transitions on the alert and it solves two purposes here, to show the history of the alert status and internally to help to index by these fields.